config/config.go

```diff
@@ -62,26 +88,259 @@ func TryLoadRootConfig(fromdir string) (cfg hcl.Config, configpath string, found
+
+	return stacks
+}
+
+func (l List) Len() int           { return len(l) }
+func (l List) Less(i, j int) bool { return l[i].dir < l[j].dir }
+func (l List) Swap(i, j int)      { l[i], l[j] = l[j], l[i] }
+
+func loadTree(rootdir string, cfgdir string, rootcfg *hcl.Config) (*Tree, error) {
+	logger := log.With().
+		Str("action", "config.loadTree()").
 		Str("dir", rootdir).
 		Logger()
 
-	cfg, err := hcl.ParseDir(rootdir, dir)
+	tree := NewTree(cfgdir)
+	if rootcfg != nil {
+		tree.Node = *rootcfg
+	} else {
+		cfg, err := hcl.ParseDir(rootdir, cfgdir)
+		if err != nil {
+			return nil, err
+		}
+		tree.Node = cfg
+	}
+
+	f, err := os.Open(cfgdir)
 	if err != nil {
-		logger.Trace().Err(err).Msg("unable to parse config, not a stack")
-		return false, err
+		return nil, errors.E(err, "failed to open rootdir directory")
+	}
+
+	logger.Trace().Msg("reading directory file names")
+
+	names, err := f.Readdirnames(0)
+	if err != nil {
+		return nil, errors.E(err, "failed to read files in %s", rootdir)
+	}
+	for _, name := range names {
+		logger = logger.With().
+			Str("filename", name).
+			Logger()
+
+		if ignoreFilename(name) {
+			logger.Trace().Msg("ignoring dot file")
+			continue
+		}
+		dir := filepath.Join(cfgdir, name)
+		st, err := os.Lstat(dir)
+		if err != nil {
+			return nil, errors.E(err, "failed to stat %s", dir)
+		}
+		if !st.IsDir() {
+			logger.Trace().Msg("ignoring non-directory file")
+			continue
+		}
+
+		logger.Trace().Msg("loading children tree")
+
+		node, err := LoadTree(rootdir, dir)
+		if err != nil {
+			return nil, errors.E(err, "failed to load config from %s", dir)
+		}
+
+		node.Parent = tree
+		tree.Children[name] = node
+	}
+	return tree, nil
+}
+
+// LoadSubTree loads a subtree into the current tree. The destdir
+func (tree *Tree) LoadSubTree(cfgdir project.Path) error {
+	var parent project.Path
+
+	var parentNode *Tree
+	parent = cfgdir.Dir()
+	for parent != "/" {
+		var found bool
+		parentNode, found = tree.Lookup(parent)
+		if found {
+			break
+		}
+		parent = parent.Dir()
+	}
+
+	if parentNode == nil {
+		parentNode = tree
+	}
+
+	rootdir := tree.RootDir()
+
+	relpath := strings.TrimPrefix(cfgdir.String(), parent.String())
+	relpath = strings.TrimPrefix(relpath, "/")
+	components := strings.Split(relpath, "/")
+	nextComponent := components[0]
+	subtreeDir := filepath.Join(rootdir, parent.String(), nextComponent)
+
+	node, err := LoadTree(rootdir, subtreeDir)
+	if err != nil {
+		return errors.E(err, "failed to load config from %s", subtreeDir)
+	}
+
+	if node.Dir() == rootdir {
+		// root configuration reloaded
+		*tree = *node
+	} else {
+		node.Parent = parentNode
+		parentNode.Children[nextComponent] = node
+	}
+	return nil
+}
+
+// IsEmptyConfig tells if the configuration is empty.
+func (tree *Tree) IsEmptyConfig() bool {
+	return tree.Node.IsEmpty()
+}
+
+// IsStack returns true if the given directory is a stack, false otherwise.
+func IsStack(cfg *Tree, dir string) bool {
```

Comment:
```suggestion
// IsStack returns true if the given directory is a stack, false otherwise.
// The given dir is an host absolute path.
func IsStack(cfg *Tree, dir string) bool {
```